### DISCUSSION TITLE
Can custom floats work with compact syntax?

### DISCUSSION THREAD

**Discussion by d-morrison** [quarto-dev/quarto-cli#11394] [Q&A] [answered]
### Description

I'm wondering whether there's a way to make the following qmd add the `Figure S#:` label to the first caption:

````qmd
---
title: "test"
format: html
crossref:
  custom:
    - kind: float
      key: suppfig
      reference-prefix: Figure S
      space-before-numbering: false
---


```{r}
#| label: suppfig-fig1
#| fig-cap: "a figure"
plot(1:10)
```

::: {#suppfig-fig2}

```{r}
#| fig-cap: "another figure"
plot(10:1)
```

:::

````

Currently, the first figure doesn't get a caption label:

![image](https://github.com/user-attachments/assets/7cffc8d6-69f2-44bc-9323-70375ae69779)

**Comment by cscheid** [comment, marked as answer]
It won't be an easy thing to change, because it will require us to teach knitr about custom crossrefs. The syntax we would support would be something like

````
```{r}
#| label: suppfig-fig1
#| suppfig-cap: "a figure"
plot(1:10)
```
````

But don't hold your breath, this is a harder change than it appears.

**Reply by cderv** [reply]
Issue to follow on this is this one 
- https://github.com/quarto-dev/quarto-cli/issues/7665
